src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -192,60 +252,125 @@ public WindowBuildTool(@Nullable final BlockPos pos)
         if (structure != null)
         {
             rotation = Settings.instance.getRotation();
-            level = Settings.instance.getLevel();
         }
         else if (pos != null)
         {
             this.pos = pos;
-            Settings.instance.pos = pos;
+            Settings.instance.setPosition(pos);
             Settings.instance.setRotation(0);
         }
 
+        initBuildingTypeNavigation();
+        initStyleNavigation();
+        initSchematicNavigation();
+
         //Register all necessary buttons with the window.
-        registerButton(BUTTON_TYPE_ID, this::placementModeClicked);
-        registerButton(BUTTON_HUT_DEC_ID, this::hutDecClicked);
-        registerButton(BUTTON_STYLE_ID, this::styleClicked);
-        registerButton(BUTTON_LEVEL_ID, this::levelClicked);
         registerButton(BUTTON_CONFIRM, this::confirmClicked);
         registerButton(BUTTON_CANCEL, this::cancelClicked);
         registerButton(BUTTON_LEFT, this::moveLeftClicked);
+        registerButton(BUTTON_MIRROR, WindowBuildTool::mirror);
         registerButton(BUTTON_RIGHT, this::moveRightClicked);
-        registerButton(BUTTON_BACK, this::moveBackClicked);
+        registerButton(BUTTON_BACKWARD, this::moveBackClicked);
         registerButton(BUTTON_FORWARD, this::moveForwardClicked);
         registerButton(BUTTON_UP, WindowBuildTool::moveUpClicked);
         registerButton(BUTTON_DOWN, WindowBuildTool::moveDownClicked);
         registerButton(BUTTON_ROTATE_RIGHT, this::rotateRightClicked);
         registerButton(BUTTON_ROTATE_LEFT, this::rotateLeftClicked);
-        registerButton(BUTTON_MIRROR, WindowBuildTool::mirror);
 
+        registerButton(BUTTON_RENAME, this::renameClicked);
+        registerButton(BUTTON_DELETE, this::deleteClicked);
+        renameButton = findPaneOfTypeByID(BUTTON_RENAME, Button.class);
+        deleteButton = findPaneOfTypeByID(BUTTON_DELETE, Button.class);
     }
 
     /**
-     * Check if the player inventory has a certain hut.
-     *
-     * @param inventory the player inventory.
-     * @param hut       the hut.
-     * @return true if so.
+     * Initialise the previous/next and drop down list for section.
      */
-    private static boolean inventoryHasHut(@NotNull final InventoryPlayer inventory, final String hut)
+    private void initBuildingTypeNavigation()
     {
-        return inventory.hasItemStack(new ItemStack(Block.getBlockFromName(Constants.MOD_ID + HUT_PREFIX + hut)));
+        registerButton(BUTTON_PREVIOUS_TYPE_ID, this::previousSection);
+        registerButton(BUTTON_NEXT_TYPE_ID, this::nextSection);
+        sectionsDropDownList = findPaneOfTypeByID(DROPDOWN_TYPE_ID, DropDownList.class);
+        sectionsDropDownList.setHandler((c,i) -> onDropDownListChanged(c,i));
+        sectionsDropDownList.setDataProvider(new DropDownList.DataProvider()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 21 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)